src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -1036,6 +1043,10 @@ public async Task ReadResponseHeadersAsync(CancellationToken cancellationToken)
                     MoveTrailersToResponseMessage(_response);
                     responseContent.SetStream(EmptyReadStream.Instance);
                 }
+                else if (_webSocketEstablished)
+                {
+                    responseContent.SetStream(new Http2ReadWriteStream(this));
```

Comment:
We will reach this `if` only in case of `_responseProtocolState == ResponseProtocolState.Complete`. It is the same condition for both 1.1 and 2.0. If we expect status but get empty response we will fail earlier, in `TryEnsureHeaders` -> `CheckResponseBodyState`